gmso/formats/lammpsdata.py

```diff
@@ -225,3 +235,265 @@ def write_lammpsdata(topology, filename, atom_style='full'):
                 topology.sites.index(dihedral.connection_members[2])+1,
                 topology.sites.index(dihedral.connection_members[3])+1
                 ))
+
+
+def read_lammpsdata(filename, atom_style='full', unit_style='real', potential='lj'):
+    """Read in a lammps data file as a GMSO topology
+
+    Parameters
+    ----------
+    filename : str
+        LAMMPS data file
+    atom_style : str, optional, default='full'
+        Inferred atom style defined by LAMMPS
+    potential: str, optional, default='lj'
+        Potential type defined in data file
+
+    Returns
+    -------
+    top : GMSO Topology
+        A GMSO Topology object
+
+    Notes
+    -----
+    See http://lammps.sandia.gov/doc/2001/data_format.html for a full description of the LAMMPS data format.  
+    Currently only supporting LJ potential parameters.
+    """
+    # TODO: Add argument to ask if user wants to infer bond type
+    top = Topology()
+
+    # Validate 'atom_style'
+    if atom_style not in ['full']:
+        raise ValueError('Atom Style "{}" is invalid or is not currently supported'.format(
+                atom_style))
+
+    # Validate 'unit_style'
+    if unit_style not in ['real']:
+        raiseValueError('Unit Style "{}" is invalid or is not currently supported'.format(
+            unit_style))
+
+    # Parse box information
+    _get_box_coordinates(filename, unit_style, top)
+    # Parse atom type information
+    top, type_list = _get_ff_information(filename, unit_style, top)
+    # Parse atom information
+    _get_atoms(filename, top, unit_style, type_list)
+    # Parse connection (bonds, angles, dihedrals) information
+    # TODO: Add more atom styles
+    if atom_style in ['full']:
+        _get_connection(filename, top, unit_style, connection_type='bond')
+        _get_connection(filename, top, unit_style, connection_type='angle')
+
+    return top
+
+def get_units(unit_style):
+    """Get units for specific LAMMPS unit style
+    """
+    # Need separate angle units for harmonic force constant and angle
+    unit_style_dict = {
+            'real': {'mass': u.g,
+                      'distance': u.angstrom,
+                      'energy': u.kcal/u.mol,
+                      'angle_k': u.radian,
+                      'angle': u.degree,
```

Comment:
Yeah I wasn't sure about this either.  My thoughts converting the exceptions is that this isn't as explicit (as you mentioned).  This might get overhauled anyways once we get Unyt unit objects set up.